### PR TITLE
feat(stepper):adds skip feature to stepper component

### PR DESCRIPTION
**Pull request by adityajhajharia**
### What does this implement/fix? Explain your changes.

...

### Does this close any currently open issues?

...

### Any other comments?

...

### Dependent PRs/Commits

...

### Describe breaking changes, if any.

...

### Checklist

Check all those that are applicable and complete.

- [ ] Merged with latest `master` branch

### CODE REVIEW

File: core/components/molecules/stepper/Stepper.tsx [hunk 1]
```diff
@@ -80,9 +83,10 @@ export const Stepper = (props: StepperProps) => {
 
         const { label, value } = step;
 
+        const skippedIndex = skipIndexes.includes(index);
         const activeStep = active === index;
-        const completedStep = completed >= index;
-        const disabled = completed + 1 < index;
+        const completedStep = completed >= index && !(skippedIndex);
```
Comment (riyalohia) [inline core/components/molecules/stepper/Stepper.tsx]: Please remove brackets around `skippedIndex`.

File: core/components/molecules/stepper/Stepper.tsx [hunk 2]
```diff
@@ -80,9 +83,10 @@ export const Stepper = (props: StepperProps) => {
 
         const { label, value } = step;
 
+        const skippedIndex = skipIndexes.includes(index);
```
Comment (riyalohia) [inline core/components/molecules/stepper/Stepper.tsx]: Please rename `skippedIndex` to `isSkipped`.

File: core/components/molecules/stepper/__stories__/variants/StepperWithSkip.story.tsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    if (active > completed) setCompleted(active);
+    setActive(updatedActive);
+    return action(`Active Index: ${updatedActive}`)();
+  };
+
+  const onSkipHandler = () => {
+    const updatedActive = active > completed ? active + 1 : completed + 1;
+    setActive(updatedActive);
+    if (!skipIndexes.includes(active)) {
+      setSkip([...skipIndexes, active, updatedActive]);
```
Comment (riyalohia) [inline core/components/molecules/stepper/__stories__/variants/StepperWithSkip.story.tsx]: `updatedActive` should not be appended in skipIndexes array.

File: core/components/molecules/stepper/Stepper.tsx [hunk 4]
```diff
@@ -80,9 +83,10 @@ export const Stepper = (props: StepperProps) => {
...
         const activeStep = active === index;
-        const completedStep = completed >= index;
-        const disabled = completed + 1 < index;
+        const completedStep = completed >= index && !(skippedIndex);
+        const disabled = completed + 1 < index && !(skippedIndex);
```
Comment (riyalohia) [inline core/components/molecules/stepper/Stepper.tsx]: Condition for disabled should be `!activeStep && !skippedIndex && (completed + 1 < index)`

**Review by innosatyam — APPROVED** [review]

**Review by satyamyadav — APPROVED** [review]

**Pull request merged**